src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalInitiallyNeeded.tsx

```diff
@@ -114,7 +112,7 @@ export const useIsApprovalInitiallyNeeded = () => {
   ])
 
   const result = useMemo(
-    () => ({ isLoading: isFirstHopLoading || (isMultiHopTrade && isSecondHopLoading) }),
+    () => ({ isLoading: isFirstHopLoading || (Boolean(isMultiHopTrade) && isSecondHopLoading) }),
```

Comment:
cleanup: looking at `selectIsActiveQuoteMultiHop` consumptions, there doesn't seem to be any need for it to return `boolean | undefined` if we want to change that instead